From eaa2673574792358e11fafd129e27de8e15f2b09 Mon Sep 17 00:00:00 2001
From: LongX Zhang <longx.zhang@intel.com>
Date: Thu, 14 Apr 2016 16:08:41 +0800
Subject: [PATCH 40/41] SSIC: Disable D3 entry on every SSIC disconnect V2

This patch is a WA to disable D3 entry on every SSIC disconnect.
It was observed that the xHCI enters the D0i3 state before
the SSIC device (Intel modem) has a chance to fully enumerate,
as a consequence the enumeration fails.

The patch fixes it by blocking xhci runtime pm between disconnect
and enumeration end.

Comparing with the original patch, this patch has below improvement.
1) With the original patch, FFD still hits SWWDT related to xhci;
2) The new patch doesn't touch codes around did_runtime_put, which
looks quite like a workaround.
3) The new patch blocks runtime of bus->controller device (it's xhci)
instead of port_dev.
4) The original patch doesn't block s3. This new patch blocks s3
till modem enumeration finishes.
5) The new patch simplifies coding by removing 2 files.

Change-Id: I999ced971d3810ff59ddaa0dedfa5facd145b54f
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-31683
Signed-off-by: Zhong Yang <yang.zhong@intel.com>
Signed-off-by: Zhang Yanmin <yanmin.zhang@intel.com>
Reviewed-on: https://android.intel.com:443/494503
Reviewed-on: https://android.intel.com:443/504786
---
 drivers/usb/host/xhci-intel-cap.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/usb/host/xhci-intel-cap.h b/drivers/usb/host/xhci-intel-cap.h
index 44e36cb..eac98b5 100644
--- a/drivers/usb/host/xhci-intel-cap.h
+++ b/drivers/usb/host/xhci-intel-cap.h
@@ -82,5 +82,3 @@ extern void hub_intel_ssic_check_block_runtime(struct usb_device *udev);
 extern void hub_intel_ssic_check_unblock_runtime(struct usb_device *udev);
 extern void xhci_intel_pipe_sync_phystatus_quirk(struct xhci_hcd *xhci,
 						 bool enable);
-extern void xhci_intel_pipe_sync_phystatus_quirk(struct xhci_hcd *xhci,
-						 bool enable);
-- 
1.9.1

